provider/source/src/time_zones/convert.rs

```diff
@@ -257,30 +386,13 @@ macro_rules! long_short_impls {
                         Some(metazones) => metazones
                             .0
                             .iter()
-                            .filter_map(|(key, metazone)| {
-                                match meta_zone_id_data.get(key) {
-                                    Some(meta_zone_short_id) => {
-                                        metazone.$field.as_ref().and_then(type_fallback).map(
-                                            |format| (meta_zone_short_id.clone(), format.clone()),
-                                        )
-                                    }
-                                    None => {
-                                        // TODO(#1781): Remove this special case once the short id is updated in CLDR
-                                        if key == "Yukon" {
-                                            metazone.$field.as_ref().and_then(type_fallback).map(
-                                                |format| {
-                                                    const TINYSTR_YUKO: tinystr::TinyAsciiStr<4> =
-                                                        tinystr::tinystr!(4, "yuko");
-                                                    (MetazoneId(TINYSTR_YUKO), format.clone())
-                                                },
-                                            )
-                                        } else {
-                                            panic!(
-                                                "Cannot find short id of meta zone for {key:?}."
-                                            )
-                                        }
-                                    }
-                                }
+                            .filter_map(|(key, metazone)| match meta_zone_id_data.get(key) {
```

Comment:
Thought: Thanks for cleaning up the code from here to the end of the file, but it took me a couple minutes to figure out that this wasn't a behavioral change. A separate PR may have been helpful.